### PR TITLE
Docs: Update integration tests documentation

### PR DESCRIPTION
**Pull request by spinscale**
Our docs were lacking behind master/2.0 releases with all the recent test infra changes

### CODE REVIEW

File: docs/reference/testing/testing-framework.asciidoc [hunk 1]
```diff
@@ -42,25 +36,18 @@ First, you need to include the testing dependency in your project. If you use ma
...
-  </dependency>
 </dependencies>
 --------------------------------------------------
 
-Replace the elasticsearch version and the lucene versions with the current elasticsearch version and its accompanying lucene release.
-And replace the "randomizedtesting version" with the version that the current elasticsearch uses.
+Replace the elasticsearch version and the lucene versions with the corresponding elasticsearch version and its accompanying lucene release.
 
-There are already have a couple of classes, you can inherit from in your own test classes. The advantages of doing so is having already defined loggers, the whole randomized infrastructure is set up already.
+There are already have a couple of classes, you can inherit from in your own test classes. The advantages of doing so is having already defined loggers as well as the whole randomized infrastructure is set up already together with a lot of helper methods.
```
Comment (nik9000) [inline docs/reference/testing/testing-framework.asciidoc]: Please drop the comma after classes!

**Comment by nik9000** [comment]
LGTM - @dadoonet might want to have a look at how you specify that elasticsearch dependency but otherwise its fine.

File: docs/reference/testing/testing-framework.asciidoc [hunk 2]
```diff
@@ -18,18 +18,12 @@ All of the tests are run using a custom junit runner, the `RandomizedRunner` pro
 [[using-elasticsearch-test-classes]]
 === Using the elasticsearch test classes
 
-First, you need to include the testing dependency in your project. If you use maven and its `pom.xml` file, it looks like this
+First, you need to include the testing dependency in your project, along side with the elasticsearch dependency you have added anyway. If you use maven and its `pom.xml` file, it looks like this
```
Comment (clintongormley) [inline docs/reference/testing/testing-framework.asciidoc]: along with the elasticsearch dependency you have already added.

File: docs/reference/testing/testing-framework.asciidoc [hunk 3]
```diff
@@ -42,25 +36,18 @@ First, you need to include the testing dependency in your project. If you use ma
...
-    <artifactId>elasticsearch</artifactId>
-    <version>${elasticsearch.version}</version>
-    <scope>test</scope>
-  </dependency>
 </dependencies>
 --------------------------------------------------
 
-Replace the elasticsearch version and the lucene versions with the current elasticsearch version and its accompanying lucene release.
-And replace the "randomizedtesting version" with the version that the current elasticsearch uses.
+Replace the elasticsearch version and the lucene versions with the corresponding elasticsearch version and its accompanying lucene release.
```
Comment (clintongormley) [inline docs/reference/testing/testing-framework.asciidoc]: lucene versions or version?

File: docs/reference/testing/testing-framework.asciidoc (hunk 1 above)
Comment (clintongormley) [inline docs/reference/testing/testing-framework.asciidoc]: We provide a few classes that you can inherit from in your own test classes which provide:
* pre-defined loggers
* randomized testing infrastructure
* a number of helper methods

File: docs/reference/testing/testing-framework.asciidoc [hunk 4]
```diff
@@ -42,25 +36,18 @@ First, you need to include the testing dependency in your project. If you use ma
...
 
 
 [[unit-tests]]
 === unit tests
 
-In case you only need to execute a unit test, because your implementation can be isolated that well and does not require an up and running elasticsearch cluster, you can use the `ElasticsearchTestCase`. If you are testing lucene features, use `ElasticsearchLuceneTestCase` and if you are testing concrete token streams, use the `ElasticsearchTokenStreamTestCase` class. Those specific classes execute additional checks, which ensure that no resources leaks are happening, after the test has run.
+In case you only need to execute a unit test, because your implementation can be isolated that well and does not require an up and running elasticsearch cluster, you can use the `ESTestCase`. If you are testing lucene features, use `ESTestCase` and if you are testing concrete token streams, use the `ESTokenStreamTestCase` class. Those specific classes execute additional checks, which ensure that no resources leaks are happening, after the test has run.
```
Comment (clintongormley) [inline docs/reference/testing/testing-framework.asciidoc]: If your test is a well isolated unit test which doesn't need a running elasticsearch cluster, you can use the `ESTestCase`.

Also remove the comma before "which ensure"

File: docs/reference/testing/testing-framework.asciidoc [hunk 5]
```diff
@@ -68,18 +55,18 @@ In case you only need to execute a unit test, because your implementation can be
 
 These kind of tests require firing up a whole cluster of nodes, before the tests can actually be run. Compared to unit tests they are obviously way more time consuming, but the test infrastructure tries to minimize the time cost by only restarting the whole cluster, if this is configured explicitly.
 
-The class your tests have to inherit from is `ElasticsearchIntegrationTest`. As soon as you inherit, there is no need for you to start any elasticsearch nodes manually in your test anymore, though you might need to ensure that at least a certain number of nodes is up.
+The class your tests have to inherit from is `ESIntegTestCase`. As soon as you inherit, there is no need for you to start any elasticsearch nodes manually in your test anymore, though you might need to ensure that at least a certain number of nodes is up.
```
Comment (clintongormley) [inline docs/reference/testing/testing-framework.asciidoc]: By inheriting from this class, you will no longer need to start elasticsearch nodes manually in your test, although you might....

File: docs/reference/testing/testing-framework.asciidoc [hunk 6]
```diff
@@ -68,18 +55,18 @@ In case you only need to execute a unit test, because your implementation can be
...
 
 [[number-of-shards]]
 ==== number of shards
 
 The number of shards used for indices created during integration tests is randomized between `1` and `10` unless overwritten upon index creation via index settings.
-Rule of thumb is not to specify the number of shards unless needed, so that each test will use a different one all the time.
+Rule of thumb is not to specify the number of shards unless needed, so that each test will use a different one all the time. Alternatively you can override the `numberOfShards()` method. The same applies to the `numberOfReplicas()` method.
```
Comment (clintongormley) [inline docs/reference/testing/testing-framework.asciidoc]: The rule of thumb

File: docs/reference/testing/testing-framework.asciidoc (hunk 5 above)
Comment (clintongormley) [inline docs/reference/testing/testing-framework.asciidoc]: a certain number of nodes ARE up.

**Comment by spinscale** [comment]
incorporated all review comments

File: docs/reference/testing/testing-framework.asciidoc [hunk 7]
```diff
@@ -68,18 +59,18 @@ In case you only need to execute a unit test, because your implementation can be
 
 These kind of tests require firing up a whole cluster of nodes, before the tests can actually be run. Compared to unit tests they are obviously way more time consuming, but the test infrastructure tries to minimize the time cost by only restarting the whole cluster, if this is configured explicitly.
 
-The class your tests have to inherit from is `ElasticsearchIntegrationTest`. As soon as you inherit, there is no need for you to start any elasticsearch nodes manually in your test anymore, though you might need to ensure that at least a certain number of nodes is up.
+The class your tests have to inherit from is `ESIntegTestCase`. By inheriting from this class, you will no longer need to start elasticsearch nodes manually in your test, although you might need to ensure that at least a certain number of nodes are up.
```
Comment (dadoonet) [inline docs/reference/testing/testing-framework.asciidoc]: I would also add that you are also able to test the same code against an external cluster using `-Dtests.cluster=localhost:9300` parameter.

**Comment by dadoonet** [comment]
I left a small comment. It looks good to me.

**Pull request merged**